crates/store/re_log_encoding/src/decoder/mod.rs

```diff
@@ -132,15 +135,39 @@ pub fn read_options(
     Ok((CrateVersion::from_bytes(version), options))
 }
 
+enum Reader<R: std::io::Read> {
+    Raw(R),
+    Buffered(std::io::BufReader<R>),
+}
+
+impl<R: std::io::Read> std::io::Read for Reader<R> {
+    #[inline]
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        match self {
+            Self::Raw(read) => read.read(buf),
+            Self::Buffered(read) => read.read(buf),
+        }
+    }
+}
+
 pub struct Decoder<R: std::io::Read> {
     version: CrateVersion,
     compression: Compression,
-    read: R,
+    read: Reader<R>,
```

Comment:
Do we ever want to support unbuffered reads? Aren't those just strictly slower in almost all cases?

Reply:
Not all cases -- if you're reading from an array of bytes, like we do in a bunch of places, adding extra buffering is pure waste of time and space.